charts/karpenter/templates/controller/deployment.yaml

```diff
@@ -59,10 +59,25 @@ spec:
       nodeSelector:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.controller.affinity }}
       affinity:
-        {{- toYaml . | nindent 8 }}
-      {{- end }}
+        {{- if .Values.controller.affinity }}
+          {{- toYaml .Values.controller.affinity | nindent 8 }}
+        {{- else }}
+        nodeAffinity: 
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: provisioning.karpenter.sh/name
+                operator: DoesNotExist
+              - key: kubernetes.io/arch
+                operator: In
+                values:
+                - amd64
```

Comment:
Don't we currently build arm images w/ ko?

Reply:
The makefile has the flag to, but I only see the amd64 one in the public registry.

Reply:
It might be that ecr public just doesn't show the other arch ones ... the manifest appears to have the hashes listed, let me do some digging real quick.

Reply:
I tested on an arm64 instance and it does appear to pull down an arm64 variant, so I removed the arch affinity.